src/main/java/dinkplugin/SettingsManager.java

```diff
@@ -158,6 +182,103 @@ private static boolean isKillCountFilterInvalid(int varbitValue) {
+                if (json == null || json.isEmpty()) {
+                    plugin.addChatWarning("Clipboard was empty");
+                    return null;
+                }
+
+                try {
+                    return gson.<Map<String, String>>fromJson(json, new TypeToken<Map<String, String>>() {}.getType());
+                } catch (Exception e) {
+                    String warning = "Failed to parse config from clipboard";
+                    log.warn(warning, e);
+                    plugin.addChatWarning(warning);
+                    return null;
+                }
+            })
+            .thenAcceptAsync(this::handleImport)
+            .exceptionally(e -> {
+                plugin.addChatWarning("Failed to read clipboard");
+                return null;
+            });
+    }
+
+    @Synchronized
+    private void handleImport(Map<String, String> map) {
+        if (map == null) return;
+
+        AtomicInteger numUpdated = new AtomicInteger();
+        Collection<String> mergedConfigs = new TreeSet<>();
+        map.forEach((key, value) -> {
+            String prevValue = configManager.getConfiguration(CONFIG_GROUP, key);
+            String newValue;
+
+            if (WEBHOOK_CONFIG_KEYS.contains(key) || "ignoredNames".equals(key)) {
+                // special case: multi-line configs that should be merged (rather than replaced)
+                Collection<String> lines = readDelimited(prevValue).collect(Collectors.toCollection(LinkedHashSet::new));
+
+                int oldCount = lines.size();
+                long added = readDelimited(value)
+                    .map(lines::add)
+                    .filter(Boolean::booleanValue)
+                    .count();
+
+                if (added > 0) {
+                    newValue = String.join("\n", lines);
+
+                    if (oldCount == 0) {
+                        mergedConfigs.add(key);
+                    }
+                } else {
+                    newValue = null;
+                }
+            } else {
+                if (Objects.equals(value, prevValue)) {
+                    newValue = null;
+                } else {
+                    newValue = value;
+                }
+            }
+
+            if (newValue != null) {
+                configManager.setConfiguration(CONFIG_GROUP, key, newValue);
+                numUpdated.incrementAndGet();
+            }
+        });
+
+        plugin.addChatSuccess(
+            String.format(
+                "Updated %d config settings (from %d total specified in import). " +
+                    "Please close and open the plugin settings panel for these changes to be visually reflected.",
+                numUpdated.get(),
+                map.size()
+            )
+        );
+
+        if (!mergedConfigs.isEmpty()) {
+            plugin.addChatSuccess("The following settings were merged (rather than being overwritten): " + String.join(", ", mergedConfigs));
```

Comment:
Do we care that the config keys may not have the most user-friendly names?

Reply:
For example:

![image](https://user-images.githubusercontent.com/8106344/216538731-9017b6d2-6ae0-4f2c-a153-4e76edbe6b8a.png)